explorer/BUILD

```diff
@@ -86,9 +153,22 @@ file_test(
             "testdata/linked_list/typed_linked_list.carbon",
         ],
     ),
-    deps = [":file_test_common"],
 )
 
+[
+    sh_test(
+        name = "%s_%s" % (test_dir, test_target),
+        size = "small",
+        srcs = ["explorer_tests"],
+        args = [
+            "--no_file_tests",
+            "--gtest_filter=" + test_suite + ".*",
+            "--fail_if_no_tests_run",
+        ],
+    )
+    for (test_dir, test_target, test_suite) in EXPLORER_GTEST_TESTS
```

Comment:
Instead of a list comprehension and such, maybe it makes sense to write a small starlark macro for this?

I also wonder about just having a single test target and using test sharding?

Reply:
Regarding a single test target: file tests already use maximum sharding (50 shards total). So non-file-test tests would easily get mixed in and harder to directly execute. Trace versus non-trace issues would also be more difficult to see -- that's why there's already separation there. You could also end up with some flakiness because trace tests are slower to execute and may end up imbalanced, forcing test size from small to medium.

Reply:
(Resolving as this no longer applies to this PR.)